packages/nextcloud/lib/src/client.dart

```diff
@@ -19,23 +21,31 @@ class NextcloudClient extends DynamiteClient {
     this.loginName,
     this.username,
     final String? password,
+    final String? appPassword,
     final String? language,
     final AppType appType = AppType.unknown,
     final String? userAgentOverride,
     super.cookieJar,
-  })  : assert(loginName != null || username == null, 'Provide loginName instead of username or both'),
+  })  : assert(
+          ((loginName != null) == (username != null)) || (loginName != null || username == null),
+          'Provide loginName instead of username or both',
```

Comment:
So `loginName` is always needed and `username` is optional?

In this case we should make the first one non nullable.

Reply:
The username variable was used for the WebDAV endpoint but that is no longer the case. It can just be removed. loginName still has to stay nullable because you can create clients without credentials.

Reply:
Fixed it in https://github.com/provokateurin/nextcloud-neon/pull/526